pkg/util/file/file.go

```diff
@@ -52,6 +54,55 @@ func ReadLatestCheckpoint(logInfoFile string) (*util.SignedCheckpoint, error) {
 	return &checkpoint, nil
 }
 
+// ReadLatestCTSignedTreeHead reads the most recent signed tree head from the log file
+func ReadLatestCTSignedTreeHead(logInfoFile string) (*ct.SignedTreeHead, error) {
+	// Each line in the file is one signed tree head
+	file, err := os.Open(logInfoFile)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	// Read line by line and get the last line
+	scanner := bufio.NewScanner(file)
+	line := ""
+	for scanner.Scan() {
+		line = scanner.Text()
+	}
+
+	var checkpoint ct.SignedTreeHead
+	err = json.Unmarshal([]byte(strings.ReplaceAll(line, "\\n", "\n")), &checkpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return &checkpoint, nil
+}
+
+// WriteCTSignedTreeHead writes a signed tree head to a given log file
+func WriteCTSignedTreeHead(sth *ct.SignedTreeHead, logInfoFile string) error {
```

Comment:
Let's take this opportunity to stop writing all checkpoints, and just overwrite the file with the latest checkpoint. This will simplify reading as well.